ndk/hooks/useProfile.tsx

```diff
@@ -1,24 +1,31 @@
 import { useEffect, useState } from "react";
-import { NDKUser } from "@nostr-dev-kit/ndk";
+import { NDKUserProfile } from "@nostr-dev-kit/ndk";
 import { useNDK } from "ndk/NDKProvider";
+import { getCachedProfile } from "../inMemoryCacheAdapter";
 
 export function useProfile({ pubkey }: { pubkey: string }) {
   const { ndk } = useNDK();
-  const [user, setUser] = useState<NDKUser | null>(null);
+  const [profile, setProfile] = useState<NDKUserProfile | undefined>(
+    getCachedProfile(pubkey, ndk)
```

Comment:
Initializing with value from cache if available to prevent user data flashing on initial render.